tests/e2e/support/page/publicLinkPage.ts

```diff
@@ -0,0 +1,68 @@
+import { errors, Page } from 'playwright'
+import util from 'util'
+import { state } from '../../cucumber/environment/shared'
+import { PublicLink } from './files/publicLink'
+
+export class PublicLinkPage {
+  private static page: Page
+  private readonly passwordInput: string
+  private readonly passwordSubmitButton: string
+  private readonly fileUploadInput: string
+  private publicResourceSelector: string
+  private uploadedResourceSelector: string
+
+  constructor() {
+    this.passwordInput = 'input[type="password"]'
+    this.passwordSubmitButton =
+      '//*[@id="password-submit"]|//*[@id="oc-textinput-3"]/ancestor::div[contains(@class, "oc-mb-s")]/following-sibling::button'
+    this.publicResourceSelector = `[data-test-resource-name="%s"]`
+    this.uploadedResourceSelector = `//tbody/tr/td[contains(text(), "%s")]`
```

Comment:
use ids or classes if possible

Reply:
tbody and tr have no class. td class is not unique as you can see `oc-td oc-pl-rm oc-table-cell oc-table-cell-align-left oc-table-cell-align-middle oc-table-cell-width-auto`.So I didn't use classes.